src/webdav/NullResource.py

```diff
@@ -114,7 +114,9 @@ def _default_PUT_factory(self, name, typ, body):
 
         if ext == '.dtml':
             ob = DTMLDocument('', __name__=name)
-        elif typ in ('text/html', 'text/xml'):
+        elif typ in ('text/html',
+                     'text/xml',
+                     'application/vnd.zopefoundation.pagetemplate'):
```

Comment:
Alternatively, we could apply the same logic as for `.dtml`, e.g. use
```python
elif ext in ('.pt', '.zpt') or typ in (...
```

Reply:
Relying on the extension might indeed be better.

Otherwise I wonder what happens when the operating system associates `.pt` with that other mime type you found.  Does the new Zope definition win?  Does the OS definition win?  Does it matter on the order in which the two definitions get loaded?